src/main/java/org/datacleaner/kettle/settings/DataCleanerSettingsDialog.java

```diff
@@ -1,41 +1,230 @@
+
+        new Label(_shell, SWT.NONE);
+        new Label(_shell, SWT.NONE);
+
+        final Text _text = new Text(_shell, SWT.BORDER);
+        _text.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, true, false, 1, 1));
+        final Button btnBrowse = new Button(_shell, SWT.NONE);
+        btnBrowse.setText("Browse");
+
+        new Label(_shell, SWT.NONE);
+        new Label(_shell, SWT.NONE);
+
+        final Label errorLabel = new Label(_shell, SWT.NONE);
+        errorLabel.setText("The selected folder does not contain a DataCleaner installation");
+        errorLabel.setForeground(getParent().getDisplay().getSystemColor(SWT.COLOR_RED));
+        errorLabel.setVisible(false);
+        new Label(_shell, SWT.NONE);
+        
+        new Label(_shell, SWT.NONE);
+        new Label(_shell, SWT.NONE);
+
+        final Label labelEdition = new Label(_shell, SWT.NONE);
+        labelEdition.setText(EDITION);
+        new Label(_shell, SWT.NONE);
+
+        final Label labelVersion = new Label(_shell, SWT.NONE);
+        labelVersion.setText(VERSION);
+        new Label(_shell, SWT.NONE);
+
+        new Label(_shell, SWT.NONE);
+        new Label(_shell, SWT.NONE);
+
+        new Label(_shell, SWT.NONE);
+        final Button btnOK = new Button(_shell, SWT.NONE);
+        btnOK.setText("OK");
+        btnOK.addSelectionListener(new SelectionAdapter() {
+            @Override
+            public void widgetSelected(SelectionEvent e) {
+                _shell.close();
+            }
+        });
 
-        shell.pack();
-        shell.open();
+        final DataCleanerFooter footer = new DataCleanerFooter(_shell);
+        footer.setLayoutData(new GridData(SWT.FILL, SWT.CENTER, false, false, 2, 1));
+
+        btnBrowse.addSelectionListener(new SelectionAdapter() {
+            public void widgetSelected(SelectionEvent event) {
+                DirectoryDialog directoryChooser = new DirectoryDialog(_shell);
+                directoryChooser.setFilterPath(_text.getText());
+                directoryChooser.setText("DataCleaner instalation folder");
+                directoryChooser.setMessage("Select a directory");
+                String dir = directoryChooser.open();
+                if (dir != null) {
+                    _text.setText(dir);
+                    _result = dir;
+                    try {
+                        SoftwareVersion editionDetails = getEditionDetails(dir);
+                        if (editionDetails != null) {
+                            String edition = EDITION.trim() + " " + editionDetails.getName();
+                            labelEdition.setText(edition);
+                            labelVersion.setText(VERSION.trim() + " " + editionDetails.getVersion());
+                            errorLabel.setVisible(false);
+                        } else {
+                            errorLabel.setVisible(true);
+                        }
+                    } catch (IOException e) {
+
```

Comment:
Depends on the details. I'm not even sure when or how it can happen.

If it is not something the user can affect, I'd either go for logging (if it is a benign problem), or a full-on popup with scary text (if it is not).

If it's the users fault, an error label would probably be fine.